lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Constants, CacheKeys } from "../utils/Constants";
+import { AccessTokenCacheItem } from "./AccessTokenCacheItem";
+import { CacheLocation } from "../Configuration";
+import { BrowserStorage } from "./BrowserStorage";
+
+/**
+ * @hidden
+ */
+export class AuthCache extends BrowserStorage {// Singleton
+    
+    private clientId: string;
+    private rollbackEnabled: boolean;
+    
+    constructor(clientId: string, cacheLocation: CacheLocation, storeAuthStateInCookie: boolean) {
+        super(cacheLocation);
+        this.clientId = clientId;
+        // This is hardcoded to true for now. We may make this configurable in the future
+        this.rollbackEnabled = true;
+        this.migrateCacheEntries(storeAuthStateInCookie);
+    }
+
+    private migrateCacheEntries(storeAuthStateInCookie: boolean) {
+        const idTokenKey = `${CacheKeys.PREFIX}.${CacheKeys.IDTOKEN}`;
+        const clientInfoKey = `${CacheKeys.PREFIX}.${CacheKeys.CLIENT_INFO}`;
+        const errorKey = `${CacheKeys.PREFIX}.${CacheKeys.ERROR}`;
+        const errorDescKey = `${CacheKeys.PREFIX}.${CacheKeys.ERROR_DESC}`;
+        
+        const idTokenValue = super.getItem(idTokenKey);
+        const clientInfoValue = super.getItem(clientInfoKey);
+        const errorValue = super.getItem(errorKey);
+        const errorDescValue = super.getItem(errorDescKey);
+        const values = [idTokenValue, clientInfoValue, errorValue, errorDescValue];
+        const keysToMigrate = [CacheKeys.IDTOKEN, CacheKeys.CLIENT_INFO, CacheKeys.ERROR, CacheKeys.ERROR_DESC];
+
+        keysToMigrate.forEach((cacheKey, index) => this.duplicateCacheEntry(cacheKey, values[index], storeAuthStateInCookie));
+    }
+
+    private duplicateCacheEntry(newKey: string, value: string, storeAuthStateInCookie?: boolean) {
+        if (value) {
+            this.setItem(newKey, value, storeAuthStateInCookie);
+        }
+    }
+
+    // Prepend msal.<client-id> to each key
+    private generateCacheKey(key: string, addInstanceId: boolean): string {
+        try {
+            JSON.parse(key);
+            return key;
```

Comment:
I'm confused, why is this just returning `key`?

Reply:
For JSON objects as keys which implies well defined schema, we do not need any append. I added a comment and will revisit this code if it can be made simpler.

Reply:
Gotya. I think a comment here would be helpful to explain that.

Reply:
Done.